lib/lightning/work_orders.ex

```diff
@@ -290,116 +297,113 @@ defmodule Lightning.WorkOrders do
           work_order: [workflow: :edges]
         ]
       )
-      |> Repo.one()
-
-    step =
+    )
+    |> Multi.one(:step, fn %{run: run} ->
       from(s in Ecto.assoc(run, :steps),
-        where: s.id == ^step_id,
-        preload: [
-          :job,
-          :input_dataclip,
-          snapshot: [triggers: :webhook_auth_methods]
```

Comment:
Could avoid preloading the `input_dataclip` and the step was not using this snapshot, so it's also removed from the preload.

Reply:
@jyeshe I am worried about this. Are you sure it won't affect workflows that have webhook auth methods configured when someone inspects the run ?

Reply:
Hey @elias-ba , I appreciate your concern, being watchful about this core feature. Please notice that the `step.snapshot` is not used and the snapshot fetched for retry is the latest version one: https://github.com/OpenFn/lightning/blob/0ef3b933c9b595ff8c93993903f537d26e574252/lib/lightning/work_orders.ex#L345

It looks like you might have found a bug, if the requirement is not to always run the retry with the latest version of the snapshot. If that's the case, would you mind opening an investigation issue?